### PR TITLE
refactor: data-table and data-iterator

### PR DESCRIPTION
**Pull request by nekosaur**
## Description
Refactored v-data-table stuff. Now there's `<v-simple-table>` and `<v-virtual-table>` that can be used by themselves. Hopefully virtual table performance is a bit better, but who knows?

Updated a ton of documentation stuff for tables and iterator

## Motivation and Context
-

## How Has This Been Tested?
-

## Markup:

<details>

```vue
-
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/docs/src/components/doc/Parameters.vue [hunk 1]
```diff
@@ -13,9 +13,17 @@
       sort-by="name"
       :items-per-page="-1"
       class="component-parameters pa-2"
+      hide-default-footer
     >
-      <template v-slot:item="{ item }">
-        <v-flex xs12 grey lighten-2 mt-2>
+      <template v-slot:default="{ items }">
```
Comment (johnleider) [inline packages/docs/src/components/doc/Parameters.vue]:
```suggestion
      <template #default="{ items }">
```

File: packages/docs/src/data/drawerItems.json [hunk 2]
```diff
@@ -83,7 +83,7 @@
       { "text": "cards", "to": "cards" },
       { "text": "carousels", "to": "carousels" },
       { "text": "chips", "to": "chips" },
-      { "text": "dataIterator", "to": "data-iterator" },
+      { "text": "dataIterators", "to": "data-iterators" },
```
Comment (johnleider) [inline packages/docs/src/data/drawerItems.json]: The data-iterator page returns a 404

File: packages/vuetify/src/components/VDataIterator/VDataFooter.sass [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// Imports
+@import '../../styles/styles.sass'
+// @import './_variables.scss'
```
Comment (johnleider) [inline packages/vuetify/src/components/VDataIterator/VDataFooter.sass]: remove if not used

File: packages/vuetify/src/util/helpers.ts [hunk 4]
```diff
@@ -443,3 +443,18 @@ export function getSlotType<T extends boolean = false> (vm: Vue, name: string, s
...
+
+export function debounce (fn: Function, delay: number) {
+  let timeoutId = 0 as any
+  return (...args: any[]) => {
+    clearTimeout(timeoutId)
+    timeoutId = setTimeout(() => fn(...args), delay)
+  }
+}
+
+export function getPrefixedScopedSlots (prefix: string, scopedSlots: any) {
```
Comment (johnleider) [inline packages/vuetify/src/util/helpers.ts]: This feels very similar to something kael built on the toolbar branch. https://github.com/vuetifyjs/vuetify/pull/6701/files#diff-5f088b65834821e88d7fdbe935ef90c6R435

Comment (nekosaur) [reply packages/vuetify/src/util/helpers.ts]: Not really

Comment (nekosaur) [reply packages/vuetify/src/util/helpers.ts]: The `genSlots` method in VDataIterator however is pretty much the same function, and can be switched out when things are merged.

File: packages/docs/src/data/drawerItems.json (hunk 2 above)
Comment (nekosaur) [reply packages/docs/src/data/drawerItems.json]: Added a redirect

File: packages/docs/src/components/doc/Parameters.vue (hunk 1 above)
Comment (nekosaur) [reply packages/docs/src/components/doc/Parameters.vue]: What's your rationale for adding the shorthand _only_ to the default slot? My thinking with not using the shorthand was to not confuse users that might not know about it.

File: packages/vuetify/src/util/helpers.ts (hunk 4 above)
Comment (johnleider) [reply packages/vuetify/src/util/helpers.ts]: I said it feels, not that it is! :)

**Pull request merged**